x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -112,13 +112,24 @@ protected AllocatedPersistentTask createTask(long id, String type, String action
             scheduler, System::nanoTime) {
 
             @Override
-            protected void innerUpdateMapping(LongConsumer handler, Consumer<Exception> errorHandler) {
+            protected void innerUpdateMapping(long minRequiredMappingVersion, long minRequiredMetadataVersion,
+                                              LongConsumer handler, Consumer<Exception> errorHandler) {
                 Index leaderIndex = params.getLeaderShardId().getIndex();
                 Index followIndex = params.getFollowShardId().getIndex();
 
                 ClusterStateRequest clusterStateRequest = CcrRequests.metaDataRequest(leaderIndex.getName());
+                if (minRequiredMetadataVersion > 0) {
+                    clusterStateRequest.waitForMetaDataVersion(minRequiredMetadataVersion)
+                        .waitForTimeout(TimeValue.timeValueNanos(Long.MAX_VALUE));
```

Comment:
should we really wait indefinitely here? Scares me a bit. A minute should be fine here (similar to `CCR_AUTO_FOLLOW_WAIT_FOR_METADATA_TIMEOUT`, maybe to be shared in same setting).